lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import clsx from "clsx";
+import {
+    LabelPairedCircleMdRegularIcon,
+    LabelPairedCircleDotMdFillIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+} from "@deriv/quill-icons";
+import "./radio-button.scss";
+
+type TRadio = {
+    className?: string;
+    classNameInfo?: string;
+    classNameLabel?: string;
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    has_info?: boolean;
+    id?: string;
+    name?: string;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    required?: boolean;
+    value?: string;
+};
+
+const RadioButton = ({
+    children,
+    className,
+    classNameInfo,
+    classNameLabel,
+    defaultChecked,
+    disabled = false,
+    has_info,
+    id,
+    onChange,
+    value,
+    ...otherProps
+}: React.PropsWithChildren<TRadio>) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
```

Comment:
Is there a reason to set setChecked with defaultChecked on useEffect as well as initially state setup ?

Reply:
The value is changed from the outside of this component, therefore, it needs the useEffect to update it.